simulation/modules/plot.py

```diff
@@ -11,61 +13,85 @@
 @attr.s(kw_only=True)
 class PlotState(ModuleState):
     step_num: int = attr.ib(default=0)
+    time_steps: np.ndarray = attr.ib(factory=lambda: np.zeros(MAX_ARRAY_LENGTH, dtype=float))
+
+    # arrays to plot against time_steps
     fungal_burdens: np.ndarray = attr.ib(factory=lambda: np.zeros(MAX_ARRAY_LENGTH, dtype=int))
     macrophage_counts: np.ndarray = attr.ib(factory=lambda: np.zeros(MAX_ARRAY_LENGTH, dtype=int))
     neutrophil_counts: np.ndarray = attr.ib(factory=lambda: np.zeros(MAX_ARRAY_LENGTH, dtype=int))
-    time_steps: np.ndarray = attr.ib(factory=lambda: np.zeros(MAX_ARRAY_LENGTH, dtype=float))
 
 
 class Plot(Module):
     name = 'plot'
+    defaults = {'plot_output_folder': 'output/plots/'}
     StateClass = PlotState
 
+    def initialize(self, state: State):
+        self.advance(state, state.time)
```

Comment:
It's clever that you didn't duplicate any of the logic here. Good idea.

However, to guard against future changes where `advance` ever gets some additional logic that's particular to only `advance` and not `initialize`, maybe it'd be good to put all the logic into a common internal method, then have both `initialize` and `advance` call that method. It's not necessary now, but may make the code more maintainable in the future.

If you name the new method with a leading underscore, it informally marks it as "private" (i.e. internal), so people will know not to call the method from outside of this class.